stannp/stannp.go

```diff
@@ -141,12 +144,71 @@ func (s *Stannp) post(inputReader io.Reader, inputURL string) (*http.Response, *
 
 	res, err := s.client.Do(req)
 	if err != nil {
-		return nil, util.BuildError(500, fmt.Sprintf("error sending req [%+v]", req), false)
+		return nil, util.BuildError(500, fmt.Sprintf("error sending req [%+v]", req))
 	}
 
 	return res, nil
 }
 
+func (s *Stannp) BytesToPDF(data []byte) (*os.File, *util.APIError) {
+	tmpFile, err := ioutil.TempFile("", "example.*.pdf")
+	if err != nil {
+		return nil, util.BuildError(500, err.Error())
+	}
+
+	if _, writeErr := tmpFile.Write(data); writeErr != nil {
+		closeErr := tmpFile.Close()
+		if closeErr != nil {
+			return nil, util.BuildError(500, closeErr.Error())
+		}
+		return nil, util.BuildError(500, writeErr.Error())
+	}
+
+	return tmpFile, nil
+}
+
+func (s *Stannp) DownloadPDF(pdfURL string) (*letter.PDFRes, *util.APIError) {
+	if !strings.HasPrefix(pdfURL, PDFURLPrefix) {
+		return nil, util.BuildError(400, fmt.Sprintf("pdfURL must begin with [%s]. your input was [%s]", PDFURLPrefix, pdfURL))
+	}
+
+	fileURL, err := url.Parse(pdfURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+	path := fileURL.Path
+	segments := strings.Split(path, "/")
+	fileName := segments[len(segments)-1]
+
+	client := http.Client{
```

Comment:
creating a custom http client here bypasses any http client configuration which may have been set on a custom http client when creating the Stannp client.